crates/extrinsics/src/env_check.rs

```diff
@@ -0,0 +1,371 @@
+use ink_metadata::InkProject;
+use scale_info::{
+    form::PortableForm,
+    Field,
+    PortableRegistry,
+    TypeDef,
+};
+
+use anyhow::{
+    Context,
+    Result,
+};
+
+fn get_node_env_fields(registry: &PortableRegistry) -> Result<Vec<Field<PortableForm>>> {
+    let env_type = registry
+        .types
+        .iter()
+        .find(|t| t.ty.path.segments.contains(&"Environment".to_string()))
+        .context("The node does not contain `Environment` type. Are you using correct `pallet-contracts` version?")?;
```

Comment:
I wonder if we should just print a warning in this case, in order to still support older versions of `pallet-contracts`?

Reply:
Not sure how this would be implemented. We need to make the overall check optional then

Reply:
Move this check to `compare_node_env_with_contract`, print the warning then return `Ok(())`.